libs/core-ui/src/lib/components/InteractiveLegend.tsx

```diff
@@ -83,4 +88,27 @@ export class InteractiveLegend extends React.PureComponent<IInteractiveLegendPro
       </div>
     );
   }
+
+  private getColorBoxCss = (
+    index: number,
+    color?: string,
+    activated?: boolean
+  ): string => {
+    const classes = interactiveLegendStyles(activated, color);
+    // this is used as data series symbol in the side panel, the sequence needs to be consist with the sequence of symbols in ScatterUtils getScatterSymbols()
+    switch (true) {
+      case index % 5 === 0:
+        return classes.circleColorBox;
+      case index % 5 === 1:
+        return classes.squareColorBox;
+      case index % 5 === 2:
+        return classes.diamondColorBox;
+      case index % 5 === 3:
+        return classes.triangleColorBox;
+      case index % 5 === 4:
+        return classes.triangleDownColorBox;
+      default:
+        return classes.circleColorBox;
```

Comment:
Seems like the code on line 111 will never execute if index >= 0. Could you use if/else instead of switch case? 

Also, does passing true in switch case work?

Reply:
I can remove case 0 to deduplicate, and in this case, the guidance is using switch case rather than if/else as it has a few cases if I remember correctly. 
Passing true works in switch case

Reply:
Maybe you write a simple if/else. The above code doesn't look like using the switch correctly.

Reply:
changed to 
```
const modIndex = index % 5;
  switch (modIndex) {
    case 1:
      return classes.squareColorBox;
    case 2:
      return classes.diamondColorBox;
    case 3:
      return classes.triangleColorBox;
    case 4:
      return classes.triangleDownColorBox;
    default:
      return classes.circleColorBox;
  }
```

Does this help?